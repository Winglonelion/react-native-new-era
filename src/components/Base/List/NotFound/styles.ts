import { StyleSheet } from 'react-native';
import CommonHeights from 'theme/CommonHeights';
import CommonWidths from 'theme/CommonWidths';
import Colors from 'theme/colors';

const styles = StyleSheet.create({
  container: {
    minWidth: 100,
    minHeight: 100,
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: CommonWidths.res1,
  },
  textEmpty: {
    fontSize: 15,
    textAlign: 'center',
    lineHeight: CommonHeights.res18,
    letterSpacing: -0.4,
    color: Colors.lightBrown,
  },
  iconImage: {
    width: CommonWidths.res200,
    height: CommonWidths.res200,
  },
  icon: {
    color: Colors.greyBrown,
  },
});

export default styles;
