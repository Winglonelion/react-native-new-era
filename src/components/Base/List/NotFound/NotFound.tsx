import React, { memo } from 'react';
import {
  Image,
  ImageProps,
  ImageStyle,
  Text,
  TextStyle,
  View,
  ViewStyle,
} from 'react-native';
import Icon from 'react-native-vector-icons/Feather';

import styles from './styles';

export interface PropTypes {
  title?: string;
  style?: ViewStyle;
  iconStyle?: ImageStyle;
  image?: ImageProps;
  textStyle?: TextStyle;
}

const NotFound: React.FC<PropTypes> = ({
  title = '',
  style,
  iconStyle,
  image,
  textStyle,
}) => (
  <View pointerEvents="none" style={[styles.container, style]}>
    {image ? (
      <Image
        resizeMethod="resize"
        resizeMode="contain"
        source={image}
        style={[styles.iconImage, iconStyle]}
      />
    ) : (
      <Icon name="inbox" size={100} style={styles.icon} />
    )}
    <Text style={[styles.textEmpty, textStyle]}>{title}</Text>
  </View>
);

export default memo(NotFound);
