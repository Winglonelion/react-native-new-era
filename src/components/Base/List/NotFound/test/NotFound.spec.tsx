import React from 'react';
import { shallow } from 'enzyme';
import NotFound, { PropTypes } from '../NotFound';

const setup = (propOverrides: PropTypes = {} as PropTypes) => {
  const props = { ...propOverrides };
  const wrapper = shallow(<NotFound {...props} />);
  return {
    props,
    wrapper,
  };
};

describe('NotFound check snapshot', () => {
  test('should match snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
describe('NotFound check render correctly', () => {
  test('should contain 1 LottieView', () => {
    const { wrapper } = setup();
    expect(wrapper.find('LottieView')).toHaveLength(1);
  });
});
