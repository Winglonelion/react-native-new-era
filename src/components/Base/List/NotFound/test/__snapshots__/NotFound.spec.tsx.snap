// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound check snapshot should match snapshot 1`] = `
<View
  pointerEvents="none"
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "minHeight": 100,
        "minWidth": 100,
        "paddingHorizontal": 2,
      },
      undefined,
    ]
  }
>
  <Icon
    allowFontScaling={false}
    name="inbox"
    size={100}
    style={
      Object {
        "color": "#F0EDEB",
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#9C9079",
          "fontSize": 15,
          "letterSpacing": -0.4,
          "lineHeight": 29.57142857142857,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  />
</View>
`;
